src/encoded/static/components/filegallery.js

```diff
@@ -3035,17 +3035,19 @@ class FileGalleryRendererComponent extends React.Component {
         if (!this.props.altFilterDefault) {
             this.setState({ selectedFilterValue: '0' });
         }
+
         // Determine how many visualizable files there are
-        // const tempFiles = filterForVisualizableFiles(this.state.files);
-        // If the graph is hidden and there are no visualizable files, set default tab to be table and set default assembly to be 'All assemblies'
-        // if (this.props.hideGraph && tempFiles.length < 1) {
-        if (this.props.hideGraph) {
-            this.setState({ currentTab: 'tables' }, () => {
-                this.filterFiles('All assemblies', 'assembly');
+        const vizFiles = filterForVisualizableFiles(this.props.context.files);
```

Comment:
@zoldello I feel like this should use `this.props.data` or this test could be inaccurate. The files to render in the file gallery should come from `this.props.data` shouldn’t they?

Reply:
@forresttanaka  I will take a look.

Reply:
@forresttanaka Let's talk about this on Friday. As things usually go with me... It's complicated. I also need a second opinion on doing this where it is always accurate but awkward to developers and users or it  not being 100% accurate but functional and less messy.